File: backend/app/schemas/poll.py
import uuid
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict


class PollOptionBase(BaseModel):
    title: str
    description: Optional[str] = None


class PollOptionCreate(PollOptionBase):
    pass


class PollOptionRead(PollOptionBase):
    id: uuid.UUID
    poll_id: uuid.UUID
    vote_count: int = 0
    has_voted: bool = False

    model_config = ConfigDict(from_attributes=True)


class PollCreate(BaseModel):
    title: str
    category: str
    options: List[PollOptionCreate]


class PollRead(BaseModel):
    id: uuid.UUID
    trip_id: uuid.UUID
    creator_id: uuid.UUID
    title: str
    category: str
    is_open: bool
    created_at: datetime
    
    # extra data
    creator_name: str
    options: List[PollOptionRead]
    total_votes: int = 0

    model_config = ConfigDict(from_attributes=True)


class PollVoteCreate(BaseModel):
    option_id: uuid.UUID

File: frontend/src/types/poll.ts
export interface PollOption {
  id: string;
  poll_id: string;
  title: string;
  description?: string;
  vote_count: number;
  has_voted: boolean;
}

export interface Poll {
  id: string;
  trip_id: string;
  creator_id: string;
  creator_name: string;
  title: string;
  category: string;
  is_open: boolean;
  created_at: string;
  options: PollOption[];
  total_votes: number;
}

export interface CreatePollOption {
  title: string;
  description?: string;
}

export interface CreatePollPayload {
  title: string;
  category: string;
  options: CreatePollOption[];
}
